packages/dd-trace/src/config.js

```diff
@@ -163,6 +164,36 @@ class Config {
       process.env.DD_TRACE_EXPERIMENTAL_TRACEPARENT_ENABLED,
       false
     )
+    const DD_TRACE_PROPAGATION_STYLE_INJECT = coalesce(
+      options.tracePropagationStyle && options.tracePropagationStyle.inject,
+      process.env.DD_TRACE_PROPAGATION_STYLE_INJECT,
+      // TODO: change this to traceparent,datadog
+      // This can't be done as a semver-minor until tracestate support also exists
+      'datadog'
+    ).split(',').reduce((m, v) => {
+      m[v.toLowerCase()] = true
+      return m
+    }, {
+      b3: isTrue(DD_TRACE_B3_ENABLED),
+      'b3 single header': isTrue(DD_TRACE_B3_ENABLED),
+      tracecontext: isTrue(DD_TRACE_TRACEPARENT_ENABLED),
+      datadog: false
+    })
+    const DD_TRACE_PROPAGATION_STYLE_EXTRACT = coalesce(
+      options.tracePropagationStyle && options.tracePropagationStyle.extract,
+      process.env.DD_TRACE_PROPAGATION_STYLE_EXTRACT,
+      // TODO: change this to traceparent,datadog
+      // This can't be done as a semver-minor until tracestate support also exists
+      'datadog'
+    ).split(',').reduce((m, v) => {
+      m[v.toLowerCase()] = true
+      return m
+    }, {
```

Comment:
Wouldn't using an array (or a set for speed) make the code clearer? It would also keep the priority which is not the case right now.

Reply:
Apparently the order actually matters, so yes, I'll need to do that.